cmd/ci-secret-bootstrap/main_test.go

```diff
@@ -2633,3 +2634,435 @@ func (f *fakeVaultClient) ListKVRecursively(prefix string) ([]string, error) {
 func (f *fakeVaultClient) UpsertKV(_ string, _ map[string]string) error {
 	return nil
 }
+
+func TestIntegration(t *testing.T) {
+	type vaultObject struct {
+		field string
+		data  []byte
+	}
+
+	testCases := []struct {
+		id              string
+		config          secretbootstrap.Config
+		secretGetters   map[string]coreclientset.SecretsGetter
+		vaultObjects    map[string][]vaultObject
+		expectedSecrets map[string][]coreapi.Secret
+		expectedErrors  bool
+	}{
+		{
+			id: "happy case",
```

Comment:
```suggestion
			id: "Successfuly create secret from config",
```

There are other possible happy cases (e.G. `Update Secret from config`, `Create Secret from Vault User Secret`, `Update Secret from Vault User Secret`)